ui/components/component-library/banner-base/banner-base.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import { ButtonIcon } from '../button-icon';
+import { ButtonLink } from '../button-link';
+import { ICON_NAMES } from '../icon';
+import { Text } from '../text';
+import Box from '../../ui/box';
+
+import {
+  COLORS,
+  DISPLAY,
+  SIZES,
+  TEXT,
+} from '../../../helpers/constants/design-system';
+
+export const BannerBase = ({
+  className,
+  title,
+  titleProps,
+  children,
+
+  actionButtonLabel,
+  actionButtonOnClick,
+  actionButtonProps,
+  startAccessory,
+  onClose,
+  closeButtonProps,
+  ...props
+}) => {
+  return (
+    <Box
+      className={classnames('mm-banner-base', className)}
+      display={DISPLAY.INLINE_FLEX}
+      gap={2}
+      backgroundColor={COLORS.BACKGROUND_DEFAULT}
+      borderRadius={SIZES.SM}
+      padding={3}
+      {...props}
+    >
+      {startAccessory && <>{startAccessory}</>}
+
+      <div>
+        {title && (
+          <Text
+            className="mm-banner-base__title"
+            variant={TEXT.BODY_LG_MEDIUM}
+            as="h5"
+            {...titleProps}
+          >
+            {title}
+          </Text>
+        )}
+        {children && (
+          <Text
+            as={typeof children === 'object' ? 'div' : 'p'}
+            className="mm-banner-base__description"
+          >
+            {children}
+          </Text>
+        )}
+        {actionButtonLabel && (
+          <ButtonLink
+            // noPadding TODO: Use noPadding option when released
+            size={SIZES.AUTO} // TODO: Remove when noPadding is added
+            onClick={actionButtonOnClick}
+            {...actionButtonProps}
+          >
+            {actionButtonLabel}
+          </ButtonLink>
+        )}
+      </div>
+      {onClose && (
+        <ButtonIcon
+          className="mm-banner-base__close-button"
+          iconName={ICON_NAMES.CLOSE_OUTLINE}
+          size={SIZES.SM}
+          ariaLabel="Close" // TODO: i18n
+          onClick={onClose}
+          {...closeButtonProps}
+        />
+      )}
+    </Box>
+  );
+};
+
+BannerBase.propTypes = {
+  /**
+   * The title of the BannerBase
+   */
+  title: PropTypes.string,
+  /**
+   * The title of the BannerBase
```

Comment:
Update prop type description
```suggestion
   * Additional props to pass to the `Text` component used for the `title` text
```